File: app/src/main/java/com/cxw/avnight/weight/CommentDialog.java
package com.cxw.avnight.weight;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;

import android.view.KeyEvent;
import android.view.View;

import android.widget.CompoundButton;
import android.widget.EditText;

import android.widget.TextView;

import com.cxw.avnight.R;


public class CommentDialog extends Dialog implements View.OnClickListener {


    TextView tv_commit;//提交

    EditText et_comment;//评论内容

    private Context context;
    private OnCommitListener listener;

    public CommentDialog(Context context) {
        this(context, R.style.inputDialog);
        this.context = context;
    }

    public CommentDialog(Context context, int themeResId) {
        super(context, themeResId);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.comment_dialog_layout);
        et_comment = findViewById(R.id.et_comment);
        tv_commit = findViewById(R.id.tv_commit);
        initListener();
    }

    private void initListener() {
        //设置显示对话框时的返回键的监听
        this.setOnKeyListener((dialogInterface, keyCode, keyEvent) -> {
            if (keyCode == KeyEvent.KEYCODE_BACK && keyEvent.getRepeatCount() == 0)
                CommentDialog.this.cancel();
            return false;
        });
        //设置EditText内容改变的监听
        et_comment.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if (!TextUtils.isEmpty(s)) {
                    tv_commit.setClickable(true);
                } else {

                    tv_commit.setClickable(false);
                }

            }

            @Override
            public void afterTextChanged(Editable s) {
            }
        });

        tv_commit.setOnClickListener(this);//提交

    }

    public void setOnCommitListener(OnCommitListener listener) {
        this.listener = listener;
    }

    public interface OnCommitListener {

        void onCommit(EditText et, View v);//提交数据
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv_commit:
                if (null != listener) {
                    listener.onCommit(et_comment, v);
                }
                break;

        }
    }



}

File: app/src/main/java/com/cxw/avnight/weight/CommentDialog.kt
package com.cxw.avnight.weight

import android.app.Dialog

import android.content.Context
import android.os.Bundle
import android.os.Handler
import android.text.Editable
import android.text.TextUtils
import android.text.TextWatcher
import android.view.Gravity
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import com.cxw.avnight.R
import org.jetbrains.anko.toast


class CommentDialog(private val mCommentsHint: String, //点击发表，内容不为空时的回调
                    var mCommentListener: CommentListener)//提示文字
    : DialogFragment() {

    private var mCommentsEt: EditText? = null

    interface CommentListener {
        fun Comment(inputText: String)
    }


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = Dialog(context!!, R.style.CommentsDialog)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        val contentView = View.inflate(activity, R.layout.comment_dialog_layout, null)
        dialog.setContentView(contentView)
        dialog.setCanceledOnTouchOutside(true)
        val window = dialog.window
        val lp = window!!.attributes
        lp.gravity = Gravity.BOTTOM
        lp.alpha = 1f
        lp.dimAmount = 0.5f
        lp.width = WindowManager.LayoutParams.MATCH_PARENT
        window.attributes = lp
        window.addFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND)
        mCommentsEt = contentView.findViewById(R.id.dialog_comment_content)
        mCommentsEt!!.hint = mCommentsHint
        val publishTv = contentView.findViewById<TextView>(R.id.dialog_comment_send)
        mCommentsEt!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                if (s.isNotEmpty()) {
                    publishTv.setBackgroundResource(R.drawable.corners_review_cansend)
                } else {
                    publishTv.setBackgroundResource(R.drawable.corners_review_send)
                }

            }
        })

        publishTv.setOnClickListener {
            if (TextUtils.isEmpty(mCommentsEt!!.text.toString())) {
                context!!.toast("评论不能为空")
                return@setOnClickListener
            }
            mCommentListener.Comment(mCommentsEt!!.text.toString())
        }
        mCommentsEt!!.isFocusable = true
        mCommentsEt!!.isFocusableInTouchMode = true
        mCommentsEt!!.requestFocus()
        val mHandler = Handler()
        dialog.setOnDismissListener { mHandler.postDelayed({ hideSoftKeyBoard() }, 100) }
        return dialog
    }


    private fun hideSoftKeyBoard() {
        try {
            (activity!!.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager)
                    .hideSoftInputFromWindow(activity!!.currentFocus!!.windowToken,
                            InputMethodManager.HIDE_NOT_ALWAYS)
        } catch (e: NullPointerException) {

        }

    }


}
